packages/common/src/api/tan-query/utils/updateCollectionData.ts

```diff
@@ -0,0 +1,42 @@
+import { QueryClient } from '@tanstack/react-query'
+import { mergeWith } from 'lodash'
+import { getContext } from 'typed-redux-saga'
+
+import { ID, Collection } from '~/models'
+import { mergeCustomizer } from '~/store/cache/mergeCustomizer'
+
+import { getCollectionQueryKey } from '../collection/useCollection'
+
+type PartialCollectionUpdate = Partial<Collection> & { playlist_id: ID }
+
+/**
+ * Updates collections with partial data, merging with existing collection data
+ * and updating both react-query and redux stores.
```

Comment:
nit: no need to mention redux anymore right?
```suggestion
 * and updating the query cache.
```